extensions/eclipse-che-theia-dashboard/src/browser/theia-dashboard-client.ts

```diff
@@ -92,7 +92,7 @@ export class TheiaDashboardClient implements FrontendApplicationContribution {
         }
         const ideWorkspaceUrl = workspace!.links!.ide!;
 
-        return ideWorkspaceUrl.replace('/che/', '/dashboard/#/ide/che/');
+        return ideWorkspaceUrl.replace(/^https?:\/\/[^\/]+/, match => `${match}/dashboard/#/ide`);
```

Comment:
So, it should work only for https? What about http?

Reply:
The question mark in a regexp makes preceding token optional. Hence, `/^https?:\/\//` matches both `https://` and `http://`.

Reply:
Oh, sorry I see.